graph/tracer.go

```diff
@@ -0,0 +1,78 @@
+package graph
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/99designs/gqlgen/graphql"
+	"github.com/opentracing/opentracing-go"
+	"github.com/opentracing/opentracing-go/ext"
+
+	opentracingLog "github.com/opentracing/opentracing-go/log"
+)
+
+// Originally taken from public domain with an Unlicense and modified: https://github.com/99designs/gqlgen-contrib/blob/master/gqlopentracing/tracer.go
+// This is a gqlgen extension that enables tracing through opentracing for each overall operation that gqlgen encounters
+// as well as each field that is accessed
+
+type Tracer struct{}
+
+var _ interface {
+	graphql.HandlerExtension
+	graphql.OperationInterceptor
+	graphql.FieldInterceptor
+} = &Tracer{}
```

Comment:
I have never seen this syntax before but is it just saying that the &Tracer{} has to meet those interfaces?

If yes, then I would prefer to have a `NewTracer` method or something that returns the specific interface type enforcing that the thing you are returning matches that interface

Reply:
yeah that's exactly what it's saying, I'll change it to make it clearer though